### DISCUSSION TITLE
对 sh_hub_get_prev_func 函数DEBUG日志的疑问

### DISCUSSION THREAD

**Discussion by Knight-ZXW** [bytedance/android-inline-hook#24] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/11730925/202859882-f8ee7caa-8a03-40bd-a029-3a4889eff7d9.png)
我理解的这个函数是返回 proxyChain 中的前一个函数，这里的注释是不是应该是 return prev enabled  proxy,  还是我理解有误

**Comment by caikelun** [comment, marked as answer]
这里确实可能引起混淆。
SLIST是个单向链表，新的proxy函数被添加到链表头部，log中的next只是表示链表中当前节点的下一个（next）节点。
如果按照proxy函数注册到SLIST中的顺序来说，这里确实是return prev enabled proxy的意思。
